lib/std/json/static.zig

```diff
@@ -247,7 +260,7 @@ fn internalParse(
             if (.object_begin != try source.next()) return error.UnexpectedToken;
 
             var result: ?T = null;
-            var name_token: ?Token = try source.nextAllocMax(allocator, .alloc_if_needed, options.max_value_len.?);
+            var name_token: ?Token = try source.nextAllocMax(allocator, options.allocate, options.max_value_len.?);
```

Comment:
Similar to above the allocation is freed, however in this case only if a union field with the corresponding name is found in `T`. I think a call to `freeAllocated` should occur in the else branch of the inline for loop below and `.alloc_if_needed` should be used here.

Reply:
I agree that the allocation change is only needed for the strings case in the `.Pointer` prong, so I made the suggested changes. In addition, there were a few more cases (in the `field_name` `switch` before the `inline for` in the `.Union` and `.Struct` prongs) where an allocated item might not be freed due to the control flow, so I added explicit calls to `freeAllocated`.